### PR TITLE
Llama 3.2 blog

### PR DESCRIPTION
**Pull request by merveenoyan**

### CODE REVIEW

File: llama32.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+- user: lewtun
+- user: ariG23498
+- user: pcuenq
+---
+
+# Llama can now see and run on your device - welcome Llama 3.2
+
+Llama 3.2 is out! Today we welcome the next iteration of the Llama collection to Hugging Face. This time, we’re excited to collaborate with Meta on the release of multimodal and small models. Ten open-weight models (5 multimodal models and 5 text-only ones) are available on the Hub.
+
+Llama 3.2 Vision comes in two sizes: 11B for efficient deployment and development on consumer-size GPU and 90B for large-scale applications. Both versions come in base and instruction-tuned variants. In addition to the four multimodal models, Meta released a new version of Llama Guard with vision support. Llama Guard 3 is a safeguard model that can classify model inputs and generations, including detecting harmful multimodal prompts or assistant responses.
```
Comment (pcuenca) [inline llama32.md:18]:
```suggestion
Llama 3.2 Vision comes in two sizes: 11B for efficient deployment and development on consumer-size GPU, and 90B for large-scale applications. Both versions come in base and instruction-tuned variants. In addition to the four multimodal models, Meta released a new version of Llama Guard with vision support. Llama Guard 3 is a safeguard model that can classify model inputs and generations, including detecting harmful multimodal prompts or assistant responses.
```

File: llama32.md [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+Llama 3.2 also includes small text-only language models that can run on-device. They come in two new sizes (1B and 3B) with base and instruct variants, and they have strong capabilities for their sizes. There’s also a small 1B version of Llama Guard that can be deployed alongside these or the larger text models in production use cases.
+
+Among the features and integrations being released, we have:
+- [Model checkpoints on the Hub](https://huggingface.co/collections/meta-llama/llama-32-66f448ffc8c32f949b04c8cf)
+- Hugging Face Transformers and TGI integration for the Vision models
+- [Hugging Chat](https://huggingface.co/chat/) integration for Meta Llama 3.2 90B Vision Instruct
```
Comment (pcuenca) [inline llama32.md:25]: Is this confirmed? cc @gary149

File: llama32.md [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+- Inference & Deployment Integration with Inference Endpoints, Google Cloud, Amazon SageMaker & DELL Enterprise Hub
+- Fine-tuning Llama 3.2 11B Vision on a single GPU with [transformers🤗](https://github.com/huggingface/huggingface-llama-recipes/llama32.ipynb) and [TRL](https://github.com/huggingface/trl/tree/main/examples/scripts/sft_vlm.py)
```
Comment (pcuenca) [inline llama32.md:27]: Does TRL work? cc @edbeeching

Comment (merveenoyan) [reply llama32.md:27]: yes it does

File: llama32.md [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+
+## Table of contents
+
+- [What is Llama 3.2 Vision?](#what-is-llama-32-vision)
+- [Llama 3.2 license changes. Sorry, EU :(](#llama-32-license-changes-sorry-eu)
+- [What is special about Llama 3.2 1B and 3B?](#what-is-special-about-llama-32-1b-and-3b)
+- [Demo](#demo)
+- [Using Hugging Face Transformers](#using-hugging-face-transformers)
+- [Llama 3.2 1B & 3B Language Models](#llama-32-1b-3b-language-models)
+- [Llama 3.2 Vision](llama-32-vision)
```
Comment (pcuenca) [inline llama32.md:37]:
```suggestion
- [Llama 3.2 Vision](#llama-32-vision)
```

File: llama32.md [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
...d Palace in Bangkok, Thailand. The Grand Palace is a complex of buildings that served as the official residence of the King of Siam and the administrative seat of government for 150 years. It is one of the most popular tourist attractions in Bangkok and a must see for anyone visiting the city. <br> Some other attractions near the Grand Palace include:  <br> * Wat Phra Kaew: This temple is located within the Grand Palace complex and is home to the famous Emerald Buddha statue. <br>  * Wat Arum: This temple is located on the west bank of the Chao Phraya River and is known for its stunning architecture and beautiful views of the city. <br>  * Wat Plo: This temple is located near the Grand Palace and is home to a large reclining Buddha statue.  <br> * Lump hini Park: This park is located in the heart of Bangkok and offers a peaceful escape from the hustle and bustle of the city. <br>  * Jim Thompson House: This museum showcases the life and work of Jim Thompson, an American businessman who helped revitalize the Thai silk industry in the 1950 s and 1960 s.  Overall, the Grand Palace is a must- see attraction in Bangkok, and there are many other interesting places to visit in the area. |
```
Comment (pcuenca) [inline llama32.md:59]: The images are not displayed in the preview for some reason

Comment (merveenoyan) [reply llama32.md:59]: for HF new-blog link they were, IDK why

File: llama32.md [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+| MMMU (val) | 41.7 | 50.7 (CoT)      | 49.3 (zero-shot) | 60.3 (CoT)       | Micro Average Accuracy |
+| VQAv2      | 66.8 (val)       | 75.2 (test)     | 73.6 (val)       | 78.1 (test)      | Accuracy |
+| DocVQA     | 62.3 (val)       | 88.4 (test)     | 70.7 (val)       | 90.1 (test)      | ANLS |
+| AI2D       | 62.4             | 91.1            | 75.3             | 92.3             | Accuracy |
+
+We expect the text capabilities of these models to be on par with the 8B and 70B Llama 3.1 models, respectively, as our understanding is that the text models were frozen during the training of the Vision models. Hence, text benchmarks should be consistent with 8B and 70B.
+
+## Llama 3.2 license changes. Sorry, EU :(
+
+![License Change](https://huggingface.co/datasets/huggingface/release-assets/resolve/main/license_change.png)
```
Comment (pcuenca) [inline llama32.md:78]: The message that made it to prod is a bit different:

Screenshot 2024-09-25 at 20 41 00

File: llama32.md [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+
+
+Tool use is supported in both the 1B and 3B instruction-tuned models. Tools are specified by the user in a zero-shot setting (the model has no previous information about the tools developers will use). Thus, the built-in tools that were part of the Llama 3.1 models (`brave_search` and `wolfram_alpha`) are no longer available.
+
+Due to their size, these small models can be used as assistants for bigger models and perform [assisted generation](https://huggingface.co/blog/assisted-generation) (also known as speculative decoding). Here is an example of using the Llama 3.2 1B model as an assistant to the Llama 3.1 8B model. For offline use cases, please check the on-device section later in the post.
+
+## Demo
+You can experiment with the three Instruct models in the following demos:
+
+- [Hugging Chat with Llama 3.2 90B Vision Instruct](https://huggingface.co/chat/): The whole inference stack is open-source. Hugging Chat is powered by [chat-ui](https://github.com/huggingface/chat-ui) and [text-generation-inference](https://github.com/huggingface/text-generation-inference).
```
Comment (pcuenca) [inline llama32.md:121]: TBC

File: llama32.md [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+- [Gradio Space with Llama 3.2 11B Vision Instruct](https://huggingface.co/spaces/huggingface-projects/llama-3.2-vision-11B)
+- [Gradio-powered Space with Llama 3.2 3B](https://huggingface.co/spaces/huggingface-projects/llama-3.2-3B-Instruct)
+- Llama 3.2 3B running on WebGPU
```
Comment (pcuenca) [inline llama32.md:124]: missing

File: llama32.md (hunk 6 above)
Comment (osanseviero) [reply llama32.md:78]: yes let's update it

File: llama32.md [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+
+- We load the model in `bfloat16`. As mentioned above, this is the type used by the original checkpoint published by Meta, so it’s the recommended way to run to ensure the best precision or conduct evaluations. Depending on your hardware, float16 might be faster.
+
+- By default, transformers uses the same sampling parameters (temperature=0.6 and top_p=0.9) as the original meta codebase. We haven’t conducted extensive tests yet, feel free to explore!
+
+## Llama 3.2 Vision
+
+The Vision models are larger, so they require more memory to run than the small text models. For reference, the 11B Vision model takes about 10 GB of GPU RAM during inference, in 4-bit mode.
+
+The easiest way to infer with the instruction-tuned Llama Vision model is to use the built-in chat template. The inputs have `user` and `assistant` roles to indicate the conversation turns. One difference with respect to the text models is that the system role is not supported. User turns may include image-text or text-only inputs. To indicate that the input contains an image, add {"type": "image"} to the content part of the input and then pass the image data to the `processor`:
```
Comment (pcuenca) [inline llama32.md:181]:
```suggestion
The easiest way to infer with the instruction-tuned Llama Vision model is to use the built-in chat template. The inputs have `user` and `assistant` roles to indicate the conversation turns. One difference with respect to the text models is that the system role is not supported. User turns may include image-text or text-only inputs. To indicate that the input contains an image, add `{"type": "image"}` to the content part of the input and then pass the image data to the `processor`:
```

File: llama32.md (hunk 6 above)
Comment (merveenoyan) [reply llama32.md:78]: okay replaced

File: llama32.md [hunk 10]
```diff
@@ -0,0 +1,417 @@
...
+ \
+    --per_device_train_batch_size 8 \
+    --gradient_accumulation_steps 8 \
+    --output_dir Llama-3.2-11B-Vision-Instruct-sft \
+    --bf16 \
+    --torch_dtype bfloat16 \
+    --gradient_checkpointing
+```
+
+You can also check out this notebook for LoRA fine-tuning using transformers and PEFT.
```
Comment (pcuenca) [inline llama32.md:395]: Link missing.

File: llama32.md [hunk 11]
```diff
@@ -0,0 +1,417 @@
...
+
+
+## Additional Resources
+
+- [Models on the Hub](https://huggingface.co/collections/meta-llama/llama-32-66f448ffc8c32f949b04c8cf)
+- [Hugging Face Llama Recipes](https://github.com/huggingface/huggingface-llama-recipes)
+- [Open LLM Leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard)
+- [Chat demo on Hugging Chat](https://huggingface.co/chat/models/meta-llama/Meta-Llama-3.1-405b-instruct/)
+- [Meta Blog](https://ai.meta.com/blog/llama-3-2-connect-2024-vision-edge-mobile-devices/)
+
```
Comment (pcuenca) [inline llama32.md:409]:
```suggestion
- [Evaluation datasets](https://huggingface.co/collections/meta-llama/llama-32-evals-66f44b3d2df1c7b136d821f0)
```

File: llama32.md (hunk 10 above)
Comment (merveenoyan) [reply llama32.md]: added

File: llama32.md [hunk 12]
```diff
@@ -0,0 +1,417 @@
...
+
+
+## Acknowledgements
+
+Releasing such models with support and evaluations in the ecosystem would not be possible without the contributions of thousands of community members who have contributed to transformers, text-generation-inference, vllm, pytorch, LM Eval Harness, and many other projects. We want to thank the VLLM team for their contribution to this release. This release couldn't have happened without all the support of Clémentine, Alina, Elie, and Loubna for LLM evaluations, Nicolas Patry, Olivier Dehaene, and Daniël de Kok for Text Generation Inference; Lysandre, Arthur, Pavel, Edward Beeching, Amy, Benjamin, Joao, Pablo, Raushan Turganbay, Matthew Carrigan, and Joshua Lochner for transformers, transformers.js, TRL, and PEFT support; Brigitte Tousignant and Florent Daudens for communication; Nathan Sarrazin and Victor for making Llama 3.2 available in Hugging Chat; Julien, Simon, Pierric, Eliott, Lucain, Alvaro, Caleb, and Mishig from the Hub team for Hub development and features for launch.
```
Comment (pcuenca) [inline llama32.md]:
```suggestion
Releasing such models with support and evaluations in the ecosystem would not be possible without the contributions of thousands of community members who have contributed to transformers, text-generation-inference, vllm, pytorch, LM Eval Harness, and many other projects. Hat tip to the VLLM team for their help in testing and reporting issues. This release couldn't have happened without all the support of Clémentine, Alina, Elie, and Loubna for LLM evaluations, Nicolas Patry, Olivier Dehaene, and Daniël de Kok for Text Generation Inference; Lysandre, Arthur, Pavel, Edward Beeching, Amy, Benjamin, Joao, Pablo, Raushan Turganbay, Matthew Carrigan, and Joshua Lochner for transformers, transformers.js, TRL, and PEFT support; Brigitte Tousignant and Florent Daudens for communication; Nathan Sarrazin and Victor for making Llama 3.2 available in Hugging Chat; Julien, Simon, Pierric, Eliott, Lucain, Alvaro, Caleb, and Mishig from the Hub team for Hub development and features for launch.
```

**Pull request merged**